lacework/resource_lacework_integration_docker_v2.go

```diff
@@ -130,106 +108,90 @@ func resourceLaceworkIntegrationDockerV2() *schema.Resource {
 
 func resourceLaceworkIntegrationDockerV2Create(d *schema.ResourceData, meta interface{}) error {
 	lacework := meta.(*api.Client)
-
-	limitByTags := d.Get("limit_by_tag").(string)
-	if tags := castAttributeToStringSlice(d, "limit_by_tags"); len(tags) != 0 {
-		limitByTags = strings.Join(tags, ",")
+	notifications := d.Get("notifications").(bool)
+	dockerhubV2Data := api.DockerhubV2Data{
+		LimitByTag:            castAttributeToStringSlice(d, "limit_by_tags"),
+		RegistryDomain:        d.Get("registry_domain").(string),
+		NonOSPackageEval:      d.Get("non_os_package_support").(bool),
+		RegistryNotifications: &notifications,
+		Credentials: api.DockerhubV2Credentials{
+			Username: d.Get("username").(string),
+			Password: d.Get("password").(string),
+			SSL:      d.Get("ssl").(bool),
+		},
 	}
 
-	limitByLabels := d.Get("limit_by_label").(string)
-	if labels := castAttributeToStringKeyMapOfStrings(d, "limit_by_labels"); len(labels) != 0 {
-		limitByLabels = joinMapStrings(labels, ",")
+	labels := castAttributeToArrayKeyMapOfStrings(d, "limit_by_labels")
+	if len(labels) != 0 {
+		dockerhubV2Data.LimitByLabel = labels
```

Comment:
why do we need this if? shouldn't this be just
```suggestion
	dockerhubV2Data.LimitByLabel = castAttributeToArrayKeyMapOfStrings(d, "limit_by_labels")
```